src/main/java/org/commcare/util/DateRangeUtils.java

```diff
@@ -0,0 +1,81 @@
+package org.commcare.util;
+
+import org.commcare.modern.util.Pair;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+public class DateRangeUtils {
```

Comment:
[these classes are moved from Android to core](https://github.com/dimagi/commcare-android/pull/2779/commits/b2597468c634847ac0835a3ae3632c5133b7e417)